PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:10 + add:0, change matches:
-    _inherit = ['discuss.channel']
-    _inherit = ['ir.ui.menu']
-    _inherit = ['mail.activity.plan']
-    _inherit = ['mail.activity.plan.template']
-    _inherit = ['mail.alias']
-    _inherit = ['res.company']
-    _inherit = ['res.config.settings']
-    _inherit = ['res.partner']
-    _inherit = ['res.users']
-    _inherit = ["resource.resource"]

Total Changes: 30

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr/models/discuss_channel.py
+++ b/addons/hr/models/discuss_channel.py
@@ -6,7 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class DiscussChannel(models.Model):
-    _inherit = ['discuss.channel']
+    _inherit = 'discuss.channel'
 
     subscription_department_ids = fields.Many2many(
         'hr.department', string='HR Departments',

--- a/addons/hr/models/hr_contract_type.py
+++ b/addons/hr/models/hr_contract_type.py
@@ -5,6 +5,7 @@ from odoo import api, fields, models
 
 
 class HrContractType(models.Model):
+    _name = 'hr.contract.type'
     _description = 'Contract Type'
     _order = 'sequence'
 

--- a/addons/hr/models/hr_department.py
+++ b/addons/hr/models/hr_department.py
@@ -6,6 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class HrDepartment(models.Model):
+    _name = 'hr.department'
     _description = "Department"
     _inherit = ['mail.thread', 'mail.activity.mixin']
     _order = "name"

--- a/addons/hr/models/hr_departure_reason.py
+++ b/addons/hr/models/hr_departure_reason.py
@@ -6,6 +6,7 @@ from odoo.exceptions import UserError
 
 
 class HrDepartureReason(models.Model):
+    _name = 'hr.departure.reason'
     _description = "Departure Reason"
     _order = "sequence"
 

--- a/addons/hr/models/hr_employee_base.py
+++ b/addons/hr/models/hr_employee_base.py
@@ -9,6 +9,7 @@ from odoo.tools import format_time
 
 
 class HrEmployeeBase(models.AbstractModel):
+    _name = 'hr.employee.base'
     _description = "Basic Employee"
     _order = 'name'
 

--- a/addons/hr/models/hr_employee_category.py
+++ b/addons/hr/models/hr_employee_category.py
@@ -7,6 +7,7 @@ from odoo import fields, models
 
 
 class HrEmployeeCategory(models.Model):
+    _name = 'hr.employee.category'
 
     _description = "Employee Category"
 

--- a/addons/hr/models/hr_employee_public.py
+++ b/addons/hr/models/hr_employee_public.py
@@ -5,6 +5,7 @@ from odoo import api, fields, models, tools
 
 
 class HrEmployeePublic(models.Model):
+    _name = 'hr.employee.public'
     _inherit = ["hr.employee.base"]
     _description = 'Public Employee'
     _order = 'name'

--- a/addons/hr/models/hr_job.py
+++ b/addons/hr/models/hr_job.py
@@ -6,6 +6,7 @@ from odoo.addons.web_editor.tools import handle_history_divergence
 
 
 class HrJob(models.Model):
+    _name = 'hr.job'
     _description = "Job Position"
     _inherit = ['mail.thread']
     _order = 'sequence'

--- a/addons/hr/models/hr_work_location.py
+++ b/addons/hr/models/hr_work_location.py
@@ -5,6 +5,7 @@ from odoo import fields, models
 
 
 class HrWorkLocation(models.Model):
+    _name = 'hr.work.location'
     _description = "Work Location"
     _order = 'name'
 

--- a/addons/hr/models/ir_ui_menu.py
+++ b/addons/hr/models/ir_ui_menu.py
@@ -5,7 +5,7 @@ from odoo import models, api, tools
 
 
 class IrUiMenu(models.Model):
-    _inherit = ['ir.ui.menu']
+    _inherit = 'ir.ui.menu'
 
     def _load_menus_blacklist(self):
         res = super()._load_menus_blacklist()

--- a/addons/hr/models/mail_activity_plan.py
+++ b/addons/hr/models/mail_activity_plan.py
@@ -6,7 +6,7 @@ from odoo.exceptions import UserError
 
 
 class MailActivityPlan(models.Model):
-    _inherit = ['mail.activity.plan']
+    _inherit = 'mail.activity.plan'
 
     department_id = fields.Many2one(
         'hr.department', check_company=True,

--- a/addons/hr/models/mail_activity_plan_template.py
+++ b/addons/hr/models/mail_activity_plan_template.py
@@ -6,7 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class MailActivityPlanTemplate(models.Model):
-    _inherit = ['mail.activity.plan.template']
+    _inherit = 'mail.activity.plan.template'
 
     responsible_type = fields.Selection(selection_add=[
         ('coach', 'Coach'),

--- a/addons/hr/models/mail_alias.py
+++ b/addons/hr/models/mail_alias.py
@@ -5,7 +5,7 @@ from odoo import fields, models, _
 
 
 class MailAlias(models.Model):
-    _inherit = ['mail.alias']
+    _inherit = 'mail.alias'
 
     alias_contact = fields.Selection(selection_add=[
         ('employees', 'Authenticated Employees'),

--- a/addons/hr/models/models.py
+++ b/addons/hr/models/models.py
@@ -6,7 +6,7 @@ from odoo.addons.mail.tools.alias_error import AliasError
 
 
 class Base(models.AbstractModel):
-    _inherit = ['base']
+    _inherit = 'base'
 
     def _alias_get_error(self, message, message_dict, alias):
         if alias.alias_contact == 'employees':

--- a/addons/hr/models/res_company.py
+++ b/addons/hr/models/res_company.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = ['res.company']
+    _inherit = 'res.company'
 
     hr_presence_control_email_amount = fields.Integer(string="# emails to send")
     hr_presence_control_ip_list = fields.Char(string="Valid IP addresses")

--- a/addons/hr/models/res_config_settings.py
+++ b/addons/hr/models/res_config_settings.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     resource_calendar_id = fields.Many2one(
         'resource.calendar', 'Company Working Hours',

--- a/addons/hr/models/res_partner.py
+++ b/addons/hr/models/res_partner.py
@@ -3,7 +3,7 @@ from odoo import fields, models, _
 
 
 class ResPartner(models.Model):
-    _inherit = ['res.partner']
+    _inherit = 'res.partner'
 
     employee_ids = fields.One2many(
         'hr.employee', 'work_contact_id', string='Employees', groups="hr.group_hr_user",

--- a/addons/hr/models/res_users.py
+++ b/addons/hr/models/res_users.py
@@ -76,7 +76,7 @@ HR_WRITABLE_FIELDS = [
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     def _employee_ids_domain(self):
         # employee_ids is considered a safe field and as such will be fetched as sudo.

--- a/addons/hr/models/resource.py
+++ b/addons/hr/models/resource.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResourceResource(models.Model):
-    _inherit = ["resource.resource"]
+    _inherit = "resource.resource"
 
     user_id = fields.Many2one(copy=False)
     employee_id = fields.One2many('hr.employee', 'resource_id', check_company=True, context={'active_test': False})
